### PR TITLE
Decode Morse Code 👨‍💻 👨‍💻 

### PR DESCRIPTION
**Pull request by tsheporamantso**
## Decode Morse Code 👨‍💻 👨‍💻 
In this milestone, 🗿 we managed to implement the following:
- [ ] Set up Linters. ➕
- [ ] Decode the morse code ➕ 
- [ ] Followed ruby best practices ➕
- [ ] Corrected linter errors ➕

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    '.--.' => 'p', '--.-' => 'q', '.-.' => 'r', '...' => 's', '-' => 't',
+    '..-' => 'u', '...-' => 'v', '.--' => 'w', '-..-' => 'x', '-.--' => 'y', '--..' => 'z',
+    '-----' => 0, '.----' => 1, '..---' => 2, '...--' => 3, '....-' => 4,
+    '.....' => 5, '-....' => 6, '--...' => 7, '---..' => 8, '----.' => 9, '$$$' => ' '
+  }
+  morse_arr = morse.gsub(/\s{3}/, ' $$$ ')
+  morse_arr = morse_arr.split(/\s/)
+  morse_arr.each { |n| print morse_to_ascii[n].upcase }
+  puts
+end
```
Comment (TedLivist) [inline main.rb]: - [ ] You have only one Morse decoding method, but from the project requirements, you should have three.

  - The first method for decoding one character.
  - The second for decoding a word using the first method.
  - The third is for decoding a sentence using the second method.

  Kindly implement that by splitting this method into 3 methods.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @tsheporamantso and @fickryiman,

Good job so far!
✔️ Decoding of one character
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 1 above)
Comment (tsheporamantso) [reply main.rb]: Thank you, Changes are implemented as required.

**Review by AbelG101 — APPROVED** [review]
# Hi @tsheporamantso & @fickryiman 👋, 

**Your project is complete!  There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉 🥳🥇**

## Highlights
- All methods are properly implemented ✔
- Morse code is correctly decoded ✔
- Correct use of Git Flow ✔
- Linter check is passing ✔
- Descriptive PR title with an good summary ✔

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 1 above)
Comment (fickryiman) [reply main.rb]: Hi @TedLivist,

Please check this requirement.
"Reuse methods whenever possible. Make sure that your code is DRY."

Our code is DRY.
1 methods can fulfilled all the cases.

Comment (fickryiman) [reply main.rb]: Our previous code is the best, please read all the requirements.
We resolved it by reuse methods.

**Comment by fickryiman** [comment]
Did the reviewer read this word "Reuse methods whenever possible. Make sure that your code is DRY." ????

**Comment by fickryiman** [comment]
DRY code rejected?
@TedLivist , @AbelG101

**Comment by tsheporamantso** [comment]
Guess he wanted 3 methods as required, nevertheless we are approved and the code we previously had is by far the best and DRY.
________________________________

**Comment by TedLivist** [comment]
Hi @fickryiman I see your project has been approved, you may be right that your method solves all Morse code types and is DRY. But I would like to highlight that the outlined requirement clearly stated you should have a method for each decoding, that is three methods. This requirement must be met.

Now given that the requirement asks for three methods, the DRY you speak of comes into play in a scenario where we want to decode a sentence, for example. Instead of writing the code for decoding a character, and then a word all over again, you just use the word decoder method to decode a sentence, to avoid repeating yourself.

Additionally, in my opinion, one method that has multiple responsibilities - in a more complex setting - can be hard to read and maintain in the long run. I think it would be better to separate the concerns.

**Pull request merged**